pandas/core/indexing.py

```diff
@@ -2028,17 +2028,9 @@ def _setitem_single_column(self, loc: int, value, plane_indexer) -> None:
                 # warn either
                 pass
             else:
-                warnings.warn(
-                    "In a future version, `df.iloc[:, i] = newvals` will attempt "
-                    "to set the values inplace instead of always setting a new "
-                    "array. To retain the old behavior, use either "
-                    "`df[df.columns[i]] = newvals` or, if columns are non-unique, "
-                    "`df.isetitem(i, newvals)`",
-                    FutureWarning,
-                    stacklevel=find_stack_level(),
-                )
                 # TODO: how to get future behavior?
                 # TODO: what if we got here indirectly via loc?
+                pass
```

Comment:
Simply removing the warning is not the correct approach. We have many cases that we want to warn about. You have to figure out what's going wrong there and fix the root cause

Reply:
Ok Sorry, let me do some anaylsis and comment on the Open Issue.

Reply:
Hi @phofl ,
Can you please check my comments for issue [#48673](https://github.com/pandas-dev/pandas/issues/48673#issuecomment-1336407547)

Reply:
@phofl do you think it's possible to tell when it's appropriate to raise this warning without it being noisy?

From https://github.com/pandas-dev/pandas/issues/48673#issuecomment-1331069191 , it seems @jbrockmendel was on board with just removing it

Reply:
Hm no this is tricky. But not sure if removing in 1.5.3 is a good idea either.

Reply:
could make it a DeprecationWarning?

Reply:
Sounds good

Reply:
I'd keep it as is, but no strong preference

Reply:
@phofl if you have an opinion on this, could you chime in on the issue? -> https://github.com/pandas-dev/pandas/issues/48673 (to keep the discussion about the exact course of action there)